File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/chat/MapParamCommand.java
package net.bestia.zoneserver.chat;

import net.bestia.messages.MessageApi;
import net.bestia.messages.chat.ChatMessage;
import net.bestia.messages.client.ToClientEnvelope;
import net.bestia.model.dao.MapParameterDAO;
import net.bestia.model.domain.Account;
import net.bestia.model.domain.Account.UserLevel;
import net.bestia.model.domain.MapParameter;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Objects;

/**
 * Returns the max current mapsize.
 *
 * <p>
 * /mapinfo
 * </p>
 *
 * @author Thomas Felix
 */
@Component
public class MapParamCommand extends BaseChatCommand {

  private static final Logger LOG = LoggerFactory.getLogger(MapParamCommand.class);

  private final MapParameterDAO mapParamDao;
  private final MessageApi akkaApi;

  @Autowired
  public MapParamCommand(
          MessageApi akkaApi,
          MapParameterDAO mapParamDao) {
    super(akkaApi);

    this.mapParamDao = Objects.requireNonNull(mapParamDao);
    this.akkaApi = Objects.requireNonNull(akkaApi);
  }

  @Override
  public boolean isCommand(String text) {
    return text.startsWith("/mapinfo");
  }

  @Override
  public UserLevel requiredUserLevel() {
    return UserLevel.USER;
  }

  @Override
  public void executeCommand(Account account, String text) {
    LOG.debug("Chatcommand: /mapinfo triggered by account {}.", account.getId());

    final MapParameter mapParam = mapParamDao.findFirstByOrderByIdDesc();

    if (mapParam == null) {
      LOG.warn("No map parameter found inside database.");
      final ChatMessage msg = ChatMessage.getSystemMessage(account.getId(), "No map info found in database.");
			final ToClientEnvelope envelope = new ToClientEnvelope(account.getId(), msg);
      akkaApi.send(envelope);
      return;
    }

    final ChatMessage msg = ChatMessage.getSystemMessage(account.getId(), mapParam.toDetailString());
    final ToClientEnvelope envelope = new ToClientEnvelope(account.getId(), msg);
    akkaApi.send(envelope);
  }

  @Override
  protected String getHelpText() {
    return "Usage: /mapinfo";
  }

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/chat/MapParamCommand.kt
package net.bestia.zoneserver.chat

import mu.KotlinLogging
import net.bestia.messages.chat.ChatResponse
import net.bestia.messages.client.ClientEnvelope
import net.bestia.model.map.MapParameterRepository
import net.bestia.model.account.Account
import net.bestia.model.account.AccountType
import net.bestia.zoneserver.actor.routing.MessageApi
import org.springframework.stereotype.Component

private val LOG = KotlinLogging.logger { }

/**
 * Returns the max current mapsize.
 *
 * /mapinfo
 *
 * @author Thomas Felix
 */
@Component
internal class MapParamCommand(
    messageApi: MessageApi,
    private val mapParamDao: MapParameterRepository
) : BaseChatCommand(messageApi) {

  public override val helpText: String
    get() = "Usage: /mapinfo"

  override fun isCommand(text: String): Boolean {
    return text.startsWith("/mapinfo")
  }

  override fun requiredUserLevel(): AccountType {
    return AccountType.USER
  }

  override fun executeCommand(account: Account, text: String) {
    LOG.debug("Chatcommand: /mapinfo triggered by account {}.", account.id)

    val mapParam = mapParamDao.findFirstByOrderByIdDesc()

    when (mapParam) {
      null -> {
        LOG.warn("No map parameter found inside database.")
        val msg = ChatResponse.getSystemMessage(account.id, "No map info found in database.")
        val envelope = ClientEnvelope(account.id, msg)
        messageApi.send(envelope)
      }
      else -> {
        val msg = ChatResponse.getSystemMessage(account.id, mapParam.toDetailString())
        val envelope = ClientEnvelope(account.id, msg)
        messageApi.send(envelope)
      }
    }
  }
}
